iothub/device/src/ModuleClient.cs

```diff
@@ -0,0 +1,363 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Devices.Client.Edge;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    using System;
+    using System.Collections.Generic;
+#if NETSTANDARD1_3
+    using System.Net.Http;
+#endif
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Shared;
+
+    /// <summary>
+    /// Contains methods that a device can use to send messages to and receive from the service.
+    /// </summary>
+    public sealed class ModuleClient : IDisposable
+    {
+        private readonly InternalClient internalClient;
+
+        internal ModuleClient(InternalClient internalClient)
```

Comment:
>internal [](start = 8, length = 8)

private to stay consistent with DeviceClient

Reply:
Emm, this is internal by design. It is used by the EdgeModuleClientFactory to instantiate the module client.